explainer.md

```diff
@@ -0,0 +1,207 @@
+
+### Handling Swipes
+In the absence of a Gesture API for customizing swipes, authors can use this API to handle swipes in the following ways:
+
+1. Add `disable-swipe` which means no visual transition occurs during the gesture. The site instead does a visual transition when the navigation commits (using the `popState` or `navigate` event). This could be sub-optimal for users since they can't use the previous page's preview to decide whether the navigation should occur.
+   However, it's unlikely that authors will do this unless a custom transition when the navigation commits is strictly better than the default UA transition.
+
+2. Omit `disable-swipe` which allows the UA to do a visual transition during the gesture. The author can detect this using the API described in [Detecting UA Transition](#detecting-ua-transition).
+
+### Global vs Navigation Based
+`same-doc-ua-transition` could be a global setting that applies to all same-document navigations originating from the current Document, or configured based on the source/destination URL. This depends on whether authors have custom transitions for a subset of navigations, or whether the behavior for swipes is set different based on the destination page.
+
+The author could simply update the value of the global setting based on what's in the navigation history, but the following cases are still infeasible:
+
+* The setting needs to be different for the back and forward entries.
+* Browser UI allows the user to traverse multiple entries, so back/fwd can jump to any entry in the navigation history.
+* The user navigates to a new URL by pasting in the URL bar.
+
+The proposal is to specify this using pair of URLs, which are the source/destination URLs for the navigation. This would effectively be a list of dictionaries which is 1:1 with a Document as follows:
+
+```
+[{
+   from: urlPattern("/articles/*");
+   to: urlPattern("/index");
+   same-doc-ua-transition: disable-swipe disable-atomic;
+ },
+ {
+   same-doc-ua-transition: disable-atomic;
+ }
+```
+
+Specifying the URL builds on the existing [URLPattern](https://wicg.github.io/urlpattern/) concept. Similar to the [specificty](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) concept in CSS, the list value applied should be chosen based on how narrowly scoped the rule is to a navigation. TODO: Clarify the exact algorithm for this.
+
+The destination URL must be the value before redirects. This is because the setting is applied when the swipe gesture starts, before the navigation is initiated (which will resolve redirects).
+
+### CSS API
+The following options are for a CSS based syntax for this API. The current proposal is [Option 1](#option-1).
+
+#### Option 1
+A new CSS [at-rule](https://developer.mozilla.org/en-US/docs/Web/CSS/At-rule) as described below:
+
+```css
+/* Applies to all same-document navigations from this Document */
+@ua-transition {
+  same-doc-ua-transition: disable-atomic;
```

Comment:
For css, you also need to propose a default value when "nothing is specified"

Reply:
Done.